File: Easy/pianostudioattivform/meta_pianostudioattivformsegstudview/meta_pianostudioattivformsegstudview.cs
/*
    Easy
    Copyright (C) 2020 Università degli Studi di Catania (www.unict.it)
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.
    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU General Public License for more details.
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/

﻿using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_pianostudioattivformsegstudview
{
    public class Meta_pianostudioattivformsegstudview : Meta_easydata 
	{
        public Meta_pianostudioattivformsegstudview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "pianostudioattivformsegstudview") {
				Name = "Attività formative pianificate";
			EditTypes.Add("segstud");
			ListingTypes.Add("segstud");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idcorsostudio","iddidprog","idiscrizione","idpianostudio","idpianostudioattivform","idreg"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "segstud": {
						DescribeAColumn(T, "anno", "Anno di corso", nPos++);
						DescribeAColumn(T, "attivform_title", "Attività formativa del corso", nPos++);
						DescribeAColumn(T, "attivform_scelta_title", "Attività formativa che lo studente svolgerà", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		//$GetSortings$

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/pianostudioattivform/meta_pianostudioattivformsegstudview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_pianostudioattivformsegstudview() {
        MetaData.apply(this, ["pianostudioattivformsegstudview"]);
        this.name = 'meta_pianostudioattivformsegstudview';
    }

    meta_pianostudioattivformsegstudview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_pianostudioattivformsegstudview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var objCalcFieldConfig = {};
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'segstud':
						this.describeAColumn(table, 'anno', 'Anno di corso', null, 1000, null);
						this.describeAColumn(table, 'attivform_title', 'Attività formativa Attività formativa del corso', null, 2100, -1);
						this.describeAColumn(table, 'attivform_aa', 'Identificativo Attività formativa del corso', null, 2500, 9);
						this.describeAColumn(table, 'attivformscelta_title', 'Attività formativa Attività formativa che lo studente svolgerà', null, 3100, -1);
						this.describeAColumn(table, 'attivformscelta_aa', 'Identificativo Attività formativa che lo studente svolgerà', null, 3500, 9);
						this.describeAColumn(table, 'sostenimento_data', 'Data Sostenimento', null, 4200, null);
						this.describeAColumn(table, 'sostenimento_giudizio', 'Giudizio Sostenimento', null, 4500, 50);
						this.describeAColumn(table, 'sostenimentoesito_title', 'Esito Esito Sostenimento', null, 5220, 50);
						this.describeAColumn(table, 'sostenimento_voto', 'Voto Sostenimento', 'fixed.2', 6000, null);
						this.describeAColumn(table, 'sostenimento_votosu', 'Su Sostenimento', null, 6200, null);
						this.describeAColumn(table, 'sostenimento_votolode', 'Lode Sostenimento', null, 6300, null);
//$objCalcFieldConfig_segstud$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idreg", "iddidprog", "idiscrizione", "idcorsostudio", "idpianostudio", "idattivform_scelta", "idpianostudioattivform"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			//$getSorting$

			//$describeTree$
        });

    window.appMeta.addMeta('pianostudioattivformsegstudview', new meta_pianostudioattivformsegstudview('pianostudioattivformsegstudview'));

	}());
